zapcore/write_syncer_test.go

```diff
@@ -65,6 +66,18 @@ func TestAddSyncWriter(t *testing.T) {
 	assert.NoError(t, ws.Sync(), "Unexpected error calling a no-op Sync method.")
 }
 
+func TestBufferWriter(t *testing.T) {
+	// If we pass a plain io.Writer, make sure that we still get a WriteSyncer
+	// with a no-op Sync.
+	buf := &bytes.Buffer{}
+	ws := Buffer(Buffer(AddSync(buf), 0, time.Millisecond), 0, 0)
+	requireWriteWorks(t, ws)
+	assert.Equal(t, "", buf.String(), "Unexpected log calling a no-op Write method.")
+	time.Sleep(2 * time.Millisecond)
```

Comment:
tests with `Sleep` tend to be flaky, please write the test avoiding `Sleep`.

One option is that you can pass in a custom `Writer` that signals when its' `Write` method is called

Reply:
Fixed